packages/cli-core/src/loggers/amplify_io_events_bridge.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  AmplifyIOHost,
+  AmplifyIoHostEventMessage,
+  AmplifyIoHostEventRequestMessageIoRequest,
+} from '@aws-amplify/plugin-types';
+
+/**
+ * Implements IIoHost interface of AmplifyIOHost
+ */
+export class AmplifyIOEventsBridge implements AmplifyIOHost {
+  /**
+   * a
+   */
+  constructor(
+    private readonly eventHandlers: {
+      notify?: (<T>(msg: AmplifyIoHostEventMessage<T>) => Promise<void>)[];
+      requestResponse?: (<T, U>(
+        msg: AmplifyIoHostEventRequestMessageIoRequest<T, U>
+      ) => Promise<U>)[];
+    }
+  ) {}
+
+  /**
+   * Receive amplify events and fan out notifying anyone interested.
+   */
+  notify<T>(msg: AmplifyIoHostEventMessage<T>): Promise<void> {
+    if (!this.eventHandlers.notify) {
+      return Promise.resolve();
+    }
+    const promises: Promise<void>[] = this.eventHandlers.notify?.flatMap(
+      (handler) => handler(msg)
+    );
+    return Promise.any(promises);
+  }
+
+  /**
+   * Receive amplify events and fan out asking everyone if they have a response for this request
+   */
+  async requestResponse<T, U>(
+    msg: AmplifyIoHostEventRequestMessageIoRequest<T, U>
+  ): Promise<U> {
+    if (!this.eventHandlers.requestResponse) {
+      return Promise.resolve(msg.defaultResponse);
+    }
+    const promises: Promise<U>[] = this.eventHandlers.requestResponse.flatMap(
+      (handler) => handler(msg)
+    );
+    // return the first undefined response if available, else the default response
+    const response = (await Promise.all(promises)).find((response) => response);
+    return response ?? msg.defaultResponse;
```

Comment:
This would be kind of random if there are many handlers and I don't see any example call sites.

Reply:
I was also not sure how to model that. Currently we don't even have a use case for this, i.e. there are no request response loggers. If we were to do that, we would need a runtime validation that only one subscriber is registered else fail.

Reply:
If we don't have a use case then we could just descope this api for now perhaps.

Reply:
Updated, the impl will not accept any `requestResponse` handlers